components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/ciba/CibaAuthRequestValidator.java

```diff
@@ -173,21 +183,30 @@ private void valiateTransactionContext(JWTClaimsSet claimsSet) throws CibaAuthFa
     private void validateBindingMessage(JWTClaimsSet claimsSet) throws CibaAuthFailureException {
 
         try {
-            // Validation for binding_message.
-            if ((claimsSet.getClaim(CibaConstants.BINDING_MESSAGE)) == null) {
-                // Request has claim for binding_message.
-                return;
-            }
 
-            if (StringUtils.isBlank(claimsSet.getStringClaim(CibaConstants.BINDING_MESSAGE))) {
-                // Binding_message with a blank value which is not acceptable.
-                if (log.isDebugEnabled()) {
-                    log.debug("Invalid CIBA Authentication Request made by client with clientID : " +
-                            claimsSet.getIssuer() +
-                            ".The request is with invalid  value for (binding_message).");
+            // Request has claim for binding_message(optional).
+            if (claimsSet.getClaim(CibaConstants.BINDING_MESSAGE) != null) {
+                // Validate binding message is not blank
+                String bindingMessage = claimsSet.getStringClaim(CibaConstants.BINDING_MESSAGE);
+                if (StringUtils.isBlank(bindingMessage)) {
+                    // Binding_message with a blank value which is not acceptable.
+                    if (log.isDebugEnabled()) {
+                        log.debug("Invalid CIBA Authentication Request made by client with clientID : " +
+                                claimsSet.getIssuer() +
+                                ".The request is with invalid  value for (binding_message).");
+                    }
+                    throw new CibaAuthFailureException(ErrorCodes.INVALID_BINDING_MESSAGE,
+                            "Invalid value for (binding_message).");
+                } else {
+                    // Validate binding message contain only plain text
+                    Pattern specialCharactersPattern = Pattern.compile("[^a-z0-9 ]", Pattern.CASE_INSENSITIVE);
```

Comment:
Can we compile and assign to a private static variable?
Otherwise we will compile this per request

Reply:
Re size, length the specification doesn't specify anything apart from mentioning  `A human-readable identifier or message`, `SHOULD be relatively short and use a limited set of plain text characters`. Refer https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html#auth_request

Reply:
For the moment we left the size open-ended. Any suggestions?

Reply:
Do we store this in any DB columns?

Reply:
It is a pass through IIRC and we don't persist IIRC. @KalanaDananjaya correct me if I am wrong.

Reply:
Replaced this part with above suggestion. 

A max size for a binding message not defined in the spec. The spec only mentions it should be "relatively short", therefore we decided not to inclue a max length

Reply:
@VivekVinushanth yeah it is only a passthrough